Profiler/DTXProfiler/ReactNativeProfiling/DTXReactNativeSampler.m

```diff
@@ -153,6 +159,78 @@ static void installDTXSignpostHook(JSContext* ctx)
 	};
 }
 
+static void installJSTraceEventsHook(JSContext* ctx)
+{
+	ctx[@"nativeTraceBeginSection"] = ^ (int tag, NSString* _name , id args)
+	{
+		static dispatch_once_t onceToken;
+		dispatch_once(&onceToken, ^{
+			sectionsRegularEvents = [[NSMutableArray alloc] init];
+		});
+		
+		DTXEventIdentifier identifier = __DTXProfilerMarkEventIntervalBegin(NSDate.date,@"JSSection",_name,@"",NO,YES,nil);
+		if(identifier)
+		{
+			@synchronized(sectionsRegularEvents) {
+				[sectionsRegularEvents addObject: identifier];
+			}
+		}
+	};
+	
+	ctx[@"nativeTraceEndSection"] = ^ (int tag)
+	{
+		DTXEventIdentifier identifier = [sectionsRegularEvents lastObject];
+		if(identifier)
+		{
+			__DTXProfilerMarkEventIntervalEnd(NSDate.date, identifier, DTXEventStatusCompleted, nil);
+			[sectionsRegularEvents removeLastObject];
+		}
+	};
+	
+	ctx[@"nativeTraceBeginAsyncSection"] = ^(int tag, NSString* name, int cookie)
+	{
+		static dispatch_once_t onceToken;
+		dispatch_once(&onceToken, ^{
+			asyncSections = [[NSMutableDictionary alloc] init];
+		});
+		
+		DTXEventIdentifier identifier = __DTXProfilerMarkEventIntervalBegin(NSDate.date,@"JSAsyncSection",name,@"",NO,YES,nil);
+		if(identifier)
+		{
+			@synchronized (asyncSections) {
+				[asyncSections setObject:identifier forKey:[NSNumber numberWithInt:cookie]];
+			}
+		}
+	};
+
+	ctx[@"nativeTraceEndAsyncSection"] = ^(int tag, NSString* name, int cookie)
+	{
+		@synchronized (asyncSections) {
+			NSNumber* _cookie = [NSNumber numberWithInt:cookie];
+			DTXEventIdentifier identifier = [asyncSections objectForKey:_cookie];
+			if(identifier)
+			{
+				__DTXProfilerMarkEventIntervalEnd(NSDate.date, identifier, DTXEventStatusCompleted, nil);
+				[asyncSections removeObjectForKey:_cookie];
+
+			}
+		}
+	};
+	
+	ctx[@"nativeTraceBeginLegacy"] = ^()
+	{
+		NSLog(@"nativeTraceBeginLegacy called");
```

Comment:
Are these never called from React Native?

Reply:
I ran the one app and I saw nothing about this events, I just implemented them to make sure the app doesn't crash if the JS call them.

Reply:
In the JavaScript, do they get any parameters?

Reply:
yes, but it doesn't matter becuase I don't use them , it wont crash if you call them with parameters, it will just do nothing, If that what you asking